NiceIO.cs

```diff
@@ -606,6 +622,22 @@ public NPath EnsureParentDirectoryExists()
 			var parent = Parent;
 			parent.EnsureDirectoryExists();
 			return parent;
+		}
+
+		public NPath AssertFileExists()
```

Comment:
I disagree.  I use it all the time now.  There were a number of places in il2cpp alone where we would do stuff like

var myFile = Something.Combine("Hello", World");

if(myFile.FileExists())
     throw;

OR, instead of that, there would be no exists checking code.  Presumably because it would have been ugly, so you get 

SomeMethod(Something.Combine("Hello", "World"))

and then you'd get who knows what error later on.  Maybe a process failed to start.  Or maybe something more obscure.

I don't see how AssertFileExists is any different than other API's that already exist such as EnsureDirectoryExists.  It's the same concept, only a slightly different scenario in that we are assuming someone else already created the file/directory for us.

AssertFileExists flows nicely which is what NiceIO is all about.  Breaking that flow to add your own 3 line if (Exists()) throw ruins the experience.

I don't see the point in having a Il2CppNiceIOExtensions filled with a handful general purpose helpers that didn't quite make the cut into the main NiceIO.  If some API is unique to il2cpp, I get it.  But this one is general purpose and legitimately makes a common file IO operation easier and flow more nicely when you are writing code.